packages/react-dom/src/__tests__/ReactFlightIntegration-test.js

```diff
@@ -0,0 +1,333 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+// Polyfills for test environment
+global.ReadableStream = require('@mattiasbuelens/web-streams-polyfill/ponyfill/es6').ReadableStream;
+global.TextDecoder = require('util').TextDecoder;
+
+let act;
+let Stream;
+let React;
+let ReactDOM;
+let ReactFlightDOMServer;
+let ReactFlightDOMClient;
+
+describe('ReactFlightIntegration', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    act = require('react-dom/test-utils').act;
+    Stream = require('stream');
+    React = require('react');
+    ReactDOM = require('react-dom');
+    ReactFlightDOMServer = require('react-dom/unstable-flight-server');
+    ReactFlightDOMClient = require('react-dom/unstable-flight-client');
+  });
+
+  function getTestStream() {
+    let writable = new Stream.PassThrough();
+    let readable = new ReadableStream({
+      start(controller) {
+        writable.on('data', chunk => {
+          controller.enqueue(chunk);
+        });
+        writable.on('end', () => {
+          controller.close();
+        });
+      },
+    });
+    return {
+      writable,
+      readable,
+    };
+  }
+
+  it('should resolve the root', async () => {
+    let {Suspense} = React;
+
+    // Model
+    function Text({children}) {
+      return <span>{children}</span>;
+    }
+    function HTML() {
+      return (
+        <div>
+          <Text>hello</Text>
+          <Text>world</Text>
+        </div>
+      );
+    }
+    function RootModel() {
+      return {
+        html: <HTML />,
+      };
+    }
+
+    // View
+    function Message({result}) {
+      return <p dangerouslySetInnerHTML={{__html: result.model.html}} />;
+    }
+    function App({result}) {
+      return (
+        <Suspense fallback={<h1>Loading...</h1>}>
+          <Message result={result} />
+        </Suspense>
+      );
+    }
+
+    let {writable, readable} = getTestStream();
+    ReactFlightDOMServer.pipeToNodeWritable(<RootModel />, writable);
+    let result = ReactFlightDOMClient.readFromReadableStream(readable);
+
+    let container = document.createElement('div');
+    ReactDOM.render(<App result={result} />, container);
```

Comment:
We should only use concurrent mode in these tests. There are some features that might actually need it or might need a breaking change. E.g. thinking about repurposing the second argument to function components.